pkg/ast/pipesearch/searchQueryParser.go

```diff
@@ -150,13 +155,13 @@ func updatePositionForGenEvents(aggs *QueryAggregators) {
 	}
 }
 
-func parsePipeSearch(searchText string, queryLanguage string, qid uint64) (*ASTNode, *QueryAggregators, error) {
+func parsePipeSearch(searchText string, queryLanguage string, qid uint64) (*ASTNode, *QueryAggregators, string, error) {
```

Comment:
Maybe this `indexName` being returned should be a `[]string` so we can support `index1 OR index2`, but you can keep it as `string` if this is just for the simplest use cases

Reply:
It could be but it was better to reuse `InitTableInfo` and make it synchronous with what we have in the curl request.

Reply:
addressed in the latest commit.